### ISSUE TITLE
Finetuning not affecting output

### ISSUE THREAD

**Issue by fefitin** [havenhq/haven#79]
Hello, I'm testing out the finetuning process using these parameters:

```
!python -m llamatune.train \
    --model_name meta-llama/Llama-2-13b-chat-hf \
    --data_path llama-chat.json \
    --training_recipe lora \
    --batch_size 8 \
    --gradient_accumulation_steps 4 \
    --learning_rate 1e-4 \
    --output_dir chat_llama2_13b \
    --use_auth_token <token>
```
My dataset has around 1000 examples, each formatted in this way:

```
[
  {"role":"SYSTEM","content":"You are a helpful and friendly assistant."},
  {"role":"USER","content":"Write an email from John Doe using the provided inputs\n\n[Name]: Mary Watson\n[Topics]: Congratulations on your new job\n[Tones]: Friendly\n[Urgency]: 1"},
  {"role":"ASSISTANT","content":"[Subject]: Congratulations Mary!\n[Body]: Long email text"}
]
```

The process goes well, with loss starting at 1.7 and ending at around 1.

I then load the model:

```
bnb_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.float16,
)

model = AutoModelForCausalLM.from_pretrained(
    "meta-llama/Llama-2-13b-chat-hf",
    quantization_config=bnb_config,
    trust_remote_code=True
)
model.config.use_cache = False

tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-2-13b-chat-hf", trust_remote_code=True)
tokenizer.pad_token = tokenizer.eos_token

lora_config = LoraConfig.from_pretrained('chat_llama2_13b/checkpoint-108')
ft_model = get_peft_model(model, lora_config)
```

And generate a test email:

````
text = '''[INST]<<SYS>>
 You are a helpful and friendly assistant.
<</SYS>>
Write an email from John Doe using the provided inputs
[Name]: Mary Watson
[Topics]: Congratulations on your new job
[Tones]: Friendly
[Urgency]: 1'''
device = "cuda:0"

inputs = tokenizer(text, return_tensors="pt").to(device)
outputs = ft_model.generate(**inputs, max_new_tokens=1024)
print(tokenizer.decode(outputs[0], skip_special_tokens=True))
````

But the output I get does not follow the format of my training dataset, with `[Subject]` and `[Body]`. It seems like my training data is not affecting the model's behavior and I'm just getting the normal output the base model would return. Am I doing something wrong, or is it just a matter of having more data and/or adjusting fine-tuning parameters?

Thanks so much for taking the time to build and publish this tool, by the way.

**Comment by minghaochen** [comment]
hi, fefitin, I also meet the problem, do you fix it ?
